From a22c7f26faf4bb485cd0682af8d93a76520e2998 Mon Sep 17 00:00:00 2001
From: Workers DevProd <workers-devprod@cloudflare.com>
Date: Mon, 2 Oct 2023 18:30:44 +0100
Subject: [PATCH 12/15] Support sourcemaps:

    * Recognise `wrangler://` URLs as "special", and always load them with Network.loadNetworkResource

    * Support a `text` property on the response to `Network.loadNetworkResource` to support providing a raw response, rather than a response stream

    * Enable the experimental `AUTHORED_DEPLOYED_GROUPING` and `JUST_MY_CODE` by default, for a better splitting of sourcemapped/deployed Worker code
---
 front_end/core/common/ParsedURL.ts       | 2 +-
 front_end/core/sdk/PageResourceLoader.ts | 4 +++-
 front_end/entrypoints/main/MainImpl.ts   | 2 ++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/front_end/core/common/ParsedURL.ts b/front_end/core/common/ParsedURL.ts
index 832d76c19c..968439ade4 100644
--- a/front_end/core/common/ParsedURL.ts
+++ b/front_end/core/common/ParsedURL.ts
@@ -366,7 +366,7 @@ export class ParsedURL {
     // Return special URLs as-is.
     const trimmedHref = href.trim();
     if (trimmedHref.startsWith('data:') || trimmedHref.startsWith('blob:') || trimmedHref.startsWith('javascript:') ||
-        trimmedHref.startsWith('mailto:')) {
+        trimmedHref.startsWith('mailto:') || trimmedHref.startsWith('wrangler:')) {
       return href as Platform.DevToolsPath.UrlString;
     }
 
diff --git a/front_end/core/sdk/PageResourceLoader.ts b/front_end/core/sdk/PageResourceLoader.ts
index e963b12ee2..5ce8734646 100644
--- a/front_end/core/sdk/PageResourceLoader.ts
+++ b/front_end/core/sdk/PageResourceLoader.ts
@@ -381,7 +381,9 @@ export class PageResourceLoader extends Common.ObjectWrapper.ObjectWrapper<Event
     const disableCache = Common.Settings.Settings.instance().moduleSetting('cache-disabled').get();
     const resource = await networkManager.loadNetworkResource(frameId, url, {disableCache, includeCredentials: true});
     try {
-      const content = resource.stream ? await ioModel.readToString(resource.stream) : '';
+      // @ts-expect-error Property 'text' does not exist on type 'LoadNetworkResourcePageResult'.
+      // Cloudflare custom extension to load network data without streams
+      const content = resource.stream ? await ioModel.readToString(resource.stream) : (resource.text ?? '');
       return {
         success: resource.success,
         content,
diff --git a/front_end/entrypoints/main/MainImpl.ts b/front_end/entrypoints/main/MainImpl.ts
index 978092dea5..a0d739121b 100644
--- a/front_end/entrypoints/main/MainImpl.ts
+++ b/front_end/entrypoints/main/MainImpl.ts
@@ -422,6 +422,8 @@ export class MainImpl {
       Root.Runtime.ExperimentName.NETWORK_PANEL_FILTER_BAR_REDESIGN,
       Root.Runtime.ExperimentName.FLOATING_ENTRY_POINTS_FOR_AI_ASSISTANCE,
       ...(Root.Runtime.Runtime.queryParam('isChromeForTesting') ? ['protocol-monitor'] : []),
+      Root.Runtime.ExperimentName.AUTHORED_DEPLOYED_GROUPING,
+      Root.Runtime.ExperimentName.JUST_MY_CODE,
     ]);
 
     Root.Runtime.experiments.cleanUpStaleExperiments();
-- 
2.39.5 (Apple Git-154)

